clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -313,6 +317,13 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
                 config.ignore(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
                 this.valueSerializer = valueSerializer;
             }
+
+            // load interceptors and make sure they get clientId
+            userProvidedConfigs.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
+            List<ProducerInterceptor<K, V>> interceptorList = (List) (new ProducerConfig(userProvidedConfigs)).getConfiguredInstances(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
```

Comment:
Thanks. This occurred to me after I added the comment while away from the computer and then forgot to say it here again. In an ideal world, we would detect the misconfiguration during producer creation, but that doesn't look straightforward in this case.